test/common/stats/symbol_table_impl_test.cc

```diff
@@ -575,20 +567,13 @@ TEST(SymbolTableTest, Memory) {
     }
   }
 
-  // This test only works if Memory::Stats::totalCurrentlyAllocated() works, which
-  // appears not to be the case in some tests, including asan, tsan, and mac.
-  if (Memory::Stats::totalCurrentlyAllocated() == 0) {
-    ENVOY_LOG_MISC(info,
-                   "SymbolTableTest.Memory comparison skipped due to malloc-stats returning 0.");
-  } else {
-    // Make sure we don't regress. Data as of 2019/05/29:
-    //
-    // string_mem_used:        6710912 (libc++), 7759488 (libstdc++).
-    // symbol_table_mem_used:  1726056 (3.9x) -- does not seem to depend on STL sizes.
-    EXPECT_LE(string_mem_used, 7759488);
-    EXPECT_LT(symbol_table_mem_used, string_mem_used / 3);
-    EXPECT_EQ(symbol_table_mem_used, 1726056);
-  }
+  // Make sure we don't regress. Data as of 2019/05/29:
+  //
+  // string_mem_used:        6710912 (libc++), 7759488 (libstdc++).
+  // symbol_table_mem_used:  1726056 (3.9x) -- does not seem to depend on STL sizes.
+  EXPECT_MEMORY_LE(string_mem_used, 7759488);
+  EXPECT_MEMORY_LE(symbol_table_mem_used, string_mem_used / 3);
+  EXPECT_MEMORY_EQ(symbol_table_mem_used, 1726056);
```

Comment:
Another approach might be to have `EXPECT_MEMORY_EQ` only, and do the `<=` and fudge factor computation inside the macro, to avoid having callers needing to think about this. It might be too much magic though.

Reply:
Yeah I had started to write it like that but felt like it was cleaner to make this explicit.